### PR TITLE
Typescript lottery

### PR DESCRIPTION
**Pull request by whalemer**
This is mostly just a test for typescripting a chat plugin.

### CODE REVIEW

File: server/global.d.ts [hunk 1]
```diff
@@ -12,6 +12,8 @@ import * as StreamsType from './../lib/streams';
 import * as UsersType from './users';
 import * as ChatType from './chat'
 
+import * as LotteryType from './chat-plugins/lottery';
```
Comment (Zarel) [inline server/global.d.ts]: This seems unnecessary?

File: server/chat.ts [hunk 2]
```diff
@@ -1408,9 +1408,14 @@ export const Chat = new class {
 		files.unshift('info.js');
 
 		for (const file of files) {
-			if (file.substr(-3) !== '.js') continue;
-			const plugin = require(`../server/chat-plugins/${file}`);
-
+			let plugin;
+			if (file.substr(-3) === '.ts') {
```
Comment (Zarel) [inline server/chat.ts]:
```suggestion
			if (file.endsWith('.ts')) {
```

File: server/chat.ts [hunk 3]
```diff
@@ -1408,9 +1408,14 @@ export const Chat = new class {
...
+				plugin = require(`./chat-plugins/${file.slice(0, -3)}`);
+			} else if (file.substr(-3) === '.js') {
```
Comment (Zarel) [inline server/chat.ts]:
```suggestion
			} else if (file.endsWith('.js')) {
```

**Comment by Zarel** [comment]
Looks great!

File: server/global.d.ts (hunk 1 above)
Comment (whalemer) [reply server/global.d.ts]: As far as I can tell it's necessary for typescript to recognise it, since the file's never directly included elsewhere.

Comment (TheJetOU) [reply server/global.d.ts]: I'm pretty sure that's because you did not include it in `tsconfig.json`.

Comment (Zarel) [reply server/global.d.ts]: Yeah, I would manually include it in `tsconfig` like JetOU is suggesting.

Comment (whalemer) [reply server/global.d.ts]: ...right, that works

**Pull request merged**